azure/ConsiderationsForServiceDesign.md

```diff
@@ -110,13 +112,107 @@ Another important design pattern for avoiding surprises is idempotency. An opera
+status of an operation it initiated. The flow looks like this:
+
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <img src="./relo.jpg" alt="The RELO flow"/>
+</p>
+<!-- markdownlint-enable MD033 -->
+
+1. The client sends the initial request to the resource to initiate the long-running operation.
+This initial request could be a PUT, PATCH, POST, or DELETE method.
+
+2. The resource validates the request and initiates the operation processing.
+It sends a response to client with a `200-OK` HTTP status code (or `201-Created` if the operation
+is a create operation) and a representation of the resource where the `status` field is set
+to a value indicating that the operation processing has been started.
+
+3. The client then issues a GET request to the resource to determine if the operation processing
+has completed.
+
+4. The resource responds with a representation of the resource. While the operation is still being
+processed, the status field will contain a "non-terminal" value, like `Processing`.
+
+5. After the operation processing has completed, a GET request from the client will receive a response
+where the status field contains a "terminal" value -- `Succeeded`, `Failed`, or `Canceled` --
+that indicates the result of the operation.
+
+A resource may support multiple outstanding RELO operations, where the status field of the resource
+indicates the combined status of the outstanding operations.
+If a new operation request is received when there is already a long-running operation in progress for a resource,
+the service should reject the operation if it is inconsistent with one already in progress.
+However, if the new operation is redundant or not inconsistent with the one in progress,
+for example a "reboot" operation on a VM that is in the process of rebooting, then the service should
+accept the request. The status field of the resource should then report the completion status of _both_
+operations.
+
+Note: The RELO pattern should not be used in cases where the completion status of individual operations
+may be important to users, as opposed to simply learning that an operation of the type they requested
+(e.g. create a resource with a specific name) has successfully completed.
+
+### Long-running operations with status monitor
+
+In the LRO with status monitor pattern, the status and results of the operation are encapsulated into
+a status monitor resource that is distinct from the target resource and specific to the individual
+operation request.  Here's what the status monitor LRO pattern looks like:
+
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <img src="./statmon.jpg" alt="The status monitor LRO flow"/>
+</p>
+<!-- markdownlint-enable MD033 -->
+
+1. The client sends the request to initiate the long-running operation.
+As in the RELO pattern, the initial request could be a PUT, PATCH, POST, or DELETE method.
```

Comment:
Do we want to mention idempotency here and suggest the client send the operation's ID?